specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,428 @@
+additional functionality, improved developer experience and performance optimizations.
+
+This feature provides APIs to access and manipulate app data stores.
+
+# 2. Background
+
+Windows provides data stores for packaged applications to store data by user and package family via
+WinRT APIs in the Windows.Storage and Windows.Management.Core namespaces. These API provide the
+means to create, use and manage data in these data stores.
+
+The Application Data APIs were originally introduced in Windows 8. However, as one of the earliest
+WinRT APIs they include some patterns out of step with current API practices and recommendations.
+Windows App SDK offers the opportunity to provide a new generation of APIs in line with the latest
+patterns and recommendations for an improved developer experience as well as functional enhancements
+and improved runtime efficiencies.
+
+Microsoft-internal tasks [22308619](https://task.ms/22308619),
+[30002314](https://task.ms/30002314), [39401257](https://task.ms/39401257),
+[39414843](https://task.ms/39414843), [40050113](https://task.ms/40050113) and
+[48332357](https://task.ms/48332357).
+
+# 3. Description
+
+This API supports app data stores focusing on the following scenarios:
+
+* ApplicationData and ApplicationDataManager functionality
+* *Path properties
+* Reliable access
+* Per-Machine data store
+* Unpackaged app data stores
+
+## 3.1. ApplicationData and ApplicationDataManager functionality
+
+This API provides an equivalent superset of
+[Windows.Storage.ApplicationData](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata),
+with the following notable differences:
+
+* **No roaming functionality** -- We don't provide equivalents to Windows.Storage.ApplicationData's
+      [RoamingSettings](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingsettings),
+      [RoamingStorageQuota](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingstoragequota),
+      [Version](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.version),
+      [SetVersionAsync()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.setversionasync),
+      [SignalDataChanged()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.signaldatachanged)
+      and
+      [DataChanged](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.datachanged). See [5.1. Platform Support](#51-platform-support) for more details.
+* **GetDefault() works for all packaged apps** -- Equivalent to .Current but works for all packaged
+  apps (not just callers running in an AppContainer). See
+  [3.3. Reliable access](#33-reliable-access).
+* **`GetForPackageFamily()`** -- Equivalent to Windows.Storage.ApplicationDataManager.CreateForPackageFamily().
+* **`GetForUser()`** -- Synchronous equivalent to GetForUserAsync().
+* **`GetForUnpackaged()`** -- Equivalent functionality for unpackaged apps. See
```

Comment:
Can `GetDefault()` just work for unpackaged, using some kind of hint in the resource section of the executable as the key to produce the path?

Reply:
Possibly. Interesting idea. Any prior art? In particular, any current practices doing this we could leverage? Or would it be new for new folks who wanted to use this new behavior?

Regardless, it's interesting but probably not urgent (v2?).

Reply:
See the implementation of Limited Access Features where you have to put a special field in the resource section for the executable.  https://learn.microsoft.com/en-us/windows/win32/menurc/versioninfo-resource is related but the ProductName and CompanyName are human-readable, not identifiers (and localized, often.)

Reply:
So we can define some named resources and fetch via

```
std::wstring GetPublisher()
{
    HRSRC resource{ ::FindResourceExW(nullptr, L"ApplicationData", "Publisher", 0)
    THROW_LAST_ERROR_IF_NULL(resource);
    const DWORD dataSize{ ::SizeofResource(nullptr, resource) };
    THROW_LAST_ERROR_IF(size == 0);
    auto data{ reinterpret_cast<wchar_t*>(::LoadResource(nullptr, resource));
    THROW_LAST_ERROR_IF_NULL(data);
    return std::wstring(data, dataSize / sizeofof(*data));
}
```

and likewise for "Product". Nice! Thanks Jon

Reply:
"Identity for unpackaged" is an interesting ongoing topic.

RECOMMEND: Defer this to a future date.